src/python/bot/tasks/setup.py

```diff
@@ -130,10 +130,10 @@ def _get_application_arguments(testcase, task_name):
   return app_args
 
 
-def _setup_memory_tools_environment(testcase):
+def _setup_memory_tools_environment(testcase, task_name):
   """Set up environment for various memory tools used."""
   env = testcase.get_metadata('env')
-  if not env:
+  if not env or task_name == 'minimize':
```

Comment:
nit: we should avoid dependencies on what the task is in our code. Can we remove this after a while since you already have the clear bit above?

Reply:
This is mostly for cases when we try migration script or something that calls tasks.add_task minimize to retrigger things, but it will cause older env to be used.

Reply:
Right, I just think this makes the code harder to understand and use to accomodate a very specific use case, so would prefer removing it if possible :)